inc/compatibility/elementor.php

```diff
@@ -296,20 +298,68 @@ private function get_global_color_prefix() {
 		return ( apply_filters( 'ti_wl_theme_is_localized', false ) ? __( 'Theme', 'neve' ) : 'Neve' ) . ' - ';
 	}
 
+	/**
+	 * Check if Elementor has a template on product archive or single product and store the result in a transient.
+	 */
+	public function update_elementor_templates() {
+		if ( class_exists( 'WooCommerce', false ) ) {
+			self::is_elementor_template( 'archive', 'product_archive' );
+			self::is_elementor_template( 'single', 'product' );
+		}
+	}
+
+	/**
+	 * Remove elementor template transients when a new template is added or deleted.
+	 *
+	 * This works only for product or shop template.
+	 *
+	 * @param  int      $post_id Current post id.
+	 * @param  \WP_Post $post WP_Post object.
+	 *
+	 * @return void
+	 */
+	public function clean_elementor_transients( $post_id, $post ) {
+		if ( $post->post_type !== 'elementor_library' ) {
+			return;
+		}
+
+		// Only delete transients if the user is adding or deleting a product_archive or a product template.
+		$template_type = get_post_meta( $post_id, '_elementor_template_type', true );
+		if ( ! in_array( $template_type, [ 'product', 'product-archive' ], true ) ) {
+			return;
+		}
+
+		$transient_name = 'neve_elementor_has_template_archive_product_archive';
+		if ( $template_type === 'product' ) {
+			$transient_name = 'neve_elementor_has_template_single_product';
+		}
+
+		delete_transient( $transient_name );
+	}
+
 	/**
 	 * Is the current page has an elementor template
 	 *
 	 * @param  string $location that location of the template such as single, archive etc.
 	 * @param  string $cond Template showing condition it can be product_archive, product etc.
 	 * @return bool
 	 */
-	public static function is_elementor_template( $location, $cond ) {
+	public static function is_elementor_template( $location, $cond, $force_refresh = false ) {
+		$transient_key        = 'neve_elementor_has_template_' . $location . '_' . $cond;
+		$transient_expiry_sec = HOUR_IN_SECONDS;
+		$cached_value         = get_transient( $transient_key ) == true; // cast transient (string) to bool
```

Comment:
you can use a class constant like `TRANSIENT_EXPIRY_SEC = HOUR_IN_SECONDS` instead of `$transient_expiry_sec` local variable